From 3f59b2c6add476c74c6d57e3afe55f9b424a96bc Mon Sep 17 00:00:00 2001
From: Varun Sharma <vsharma@habana.ai>
Date: Tue, 12 Apr 2022 09:52:03 +0300
Subject: [PATCH] [SW-77546] greco: mask lbw region pcie addr dec interrupt

This patch mask pcie addr dec interrupt raised due to lbw region error.
Currently its not mapped in cause register
(mmPCIE_WRAP_PCIE_IC_SEI_INTR_IND) of this interrupt and we sometimes
get this interrupt with cause 0.
After discussion with HW team , will decide how to take care of this
and then will unmask it.

Change-Id: I550cdedd5286d23934555bbbc0ea0e71c86d8a2d
Signed-off-by: Varun Sharma <vsharma@habana.ai>
---

diff --git a/arch/greco/common/common.c b/arch/greco/common/common.c
index e56f70b..f1fbf99 100644
--- a/arch/greco/common/common.c
+++ b/arch/greco/common/common.c
@@ -10,6 +10,7 @@
 #include <psoc_global_conf_regs.h>
 #include <psoc_global_conf_masks.h>
 #include <pcie_wrap_regs.h>
+#include <pcie_wrap_special_regs.h>
 #include <psoc_efuse_regs.h>
 #include <pcie_phy_regs.h>
 #include <pcie_phy_masks.h>
@@ -822,6 +823,12 @@
 	regmap_writel(reg & ~bitmask, mmPCIE_WRAP_PCIE_IC_SEI_INTR_IND);
 	/* Clear the interrupt */
 	regmap_writel(BIT(0), mmPCIE_WRAP_AXI_INTR);
+	/*
+	 * Mask PCIE ADDR DEC interrupt arising due to LBW region access.
+	 * Currently in greco, LBW region interrupts don't show in cause register.
+	 * TODO: Checking with HW team regarding this, until then mask it.
+	 */
+	regmap_writel(BIT(0), mmPCIE_WRAP_SPECIAL_GLBL_SPARE_0);
 	/* Ack PCI drain indication */
 	regmap_writel(0, mmPCIE_WRAP_AXI_DRAIN_IND);
 }
